src/SendGrid/Helpers/Mail/SendGridMessage.cs

```diff
@@ -162,34 +162,43 @@ public void AddTo(EmailAddress email, int personalizationIndex = 0, Personalizat
             if (personalization != null)
             {
                 personalization.Tos = personalization.Tos ?? new List<EmailAddress>();
-                personalization.Tos.Add(email);
-                if (this.Personalizations == null)
-                {
-                    this.Personalizations = new List<Personalization>();
-                    this.Personalizations.Add(personalization);
-                }
-                else
+
+                if (!personalization.Tos.Any(x => string.Equals(x.Email, email.Email, StringComparison.OrdinalIgnoreCase)))
```

Comment:
It would be more succinct to simply call `Contains` here to keep the way we do email comparisons uniform.

Reply:
This does a case-insensitive compare of the email address string